qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -310,23 +316,28 @@ public void startElasticsearch() throws Exception {
         return Archives.startElasticsearchWithTty(installation, sh, password);
     }
 
-
     public void assertElasticsearchFailure(Shell.Result result, String expectedMessage, Packages.JournaldWrapper journaldWrapper) {
+        assertElasticsearchFailure(result, Collections.singletonList(expectedMessage), journaldWrapper);
+    }
 
+    public void assertElasticsearchFailure(Shell.Result result, List<String> expectedMessages, Packages.JournaldWrapper journaldWrapper) {
+        @SuppressWarnings("unchecked")
+        Matcher<String>[] stringMatchers = expectedMessages.stream().map(CoreMatchers::containsString).toArray(Matcher[]::new);
         if (Files.exists(installation.logs.resolve("elasticsearch.log"))) {
 
             // If log file exists, then we have bootstrapped our logging and the
             // error should be in the logs
             assertTrue("log file exists", Files.exists(installation.logs.resolve("elasticsearch.log")));
             String logfile = FileUtils.slurp(installation.logs.resolve("elasticsearch.log"));
-            assertThat(logfile, containsString(expectedMessage));
+
+            assertThat(logfile, anyOf(Arrays.asList(stringMatchers)));
```

Comment:
Should be the following instead? The other uses of `stringMatchers` in this method aren't wrapped in `Arrays.asList`.

```suggestion
            assertThat(logfile, anyOf(stringMatchers));
```

Reply:
Both work as anyOf has multiple signatures, one that takes a vararg ( so we can pass an array of Matcher ) and it also has one that takes an Iterable ( so we can pass a List  ). That said, the Arrays.asList is a leftover from a previous attempt and we gain nothing from converting the Array to a List, so I'll change for consistency